WordPress/Classes/ViewRelated/Menus/Views/MenuItemSourceViewController.m

```diff
@@ -0,0 +1,208 @@
+#import "MenuItemSourceViewController.h"
+#import "MenuItemSourceHeaderView.h"
+#import "MenuItemPagesViewController.h"
+#import "MenuItemLinkViewController.h"
+#import "MenuItemCategoriesViewController.h"
+#import "MenuItemTagsViewController.h"
+#import "MenuItemPostsViewController.h"
+#import "Menu.h"
+
+@interface MenuItemSourceViewController () <MenuItemSourceHeaderViewDelegate, MenuItemSourceResultsViewControllerDelegate>
+
+@property (nonatomic, strong, readonly) UIStackView *stackView;
+@property (nonatomic, strong, readonly) MenuItemSourceHeaderView *headerView;
+@property (nonatomic, strong) MenuItemSourceResultsViewController *sourceViewController;
+@property (nonatomic, strong, readonly) NSCache *sourceViewControllerCache;
+@property (nonatomic, assign) BOOL itemNameWasUpdatedExternally;
+
+@end
+
+@implementation MenuItemSourceViewController
+
+- (void)awakeFromNib
+{
+    [super awakeFromNib];
+    
+    self.view.backgroundColor = [UIColor whiteColor];
+    self.view.translatesAutoresizingMaskIntoConstraints = NO;
+    _sourceViewControllerCache = [[NSCache alloc] init];
+    
+    [self setupStackView];
+    [self setupHeaderView];
+}
+
+- (void)setupStackView
+{
+    UIStackView *stackView = [[UIStackView alloc] init];
+    stackView.translatesAutoresizingMaskIntoConstraints = NO;
+    stackView.distribution = UIStackViewDistributionFill;
+    stackView.alignment = UIStackViewAlignmentFill;
+    stackView.axis = UILayoutConstraintAxisVertical;
+    stackView.spacing = 0;
+    [self.view addSubview:stackView];
+    
+    [NSLayoutConstraint activateConstraints:@[
+                                              [stackView.topAnchor constraintEqualToAnchor:self.view.topAnchor],
+                                              [stackView.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
+                                              [stackView.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
+                                              [stackView.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor],
+                                              [stackView.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor]
+                                              ]];
+    
+    _stackView = stackView;
+}
+
+- (void)setupHeaderView
+{
+    MenuItemSourceHeaderView *headerView = [[MenuItemSourceHeaderView alloc] init];
```

Comment:
For this one I was able to move the UIStackView to the storyboard. The headerView has to remain as an in-code setup due to the issues with nibs not handling stackViews properly.